vm/actor/src/builtin/market/mod.rs

```diff
@@ -476,6 +469,176 @@ impl Actor {
+
+            let mut lt = BalanceTable::from_root(rt.store(), &st.locked_table)
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+            let mut i = st.last_cron + 1;
+            while i <= rt.curr_epoch() {
+                dbe.for_each(i, |id| {
+                    let mut state: DealState = states
+                        .get(id)
+                        .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))
+                        .unwrap()
+                        .ok_or(format!("could not find deal state: {}", id))
+                        .unwrap();
+
+                    let deal = st.must_get_deal(rt.store(), id).unwrap();
+                    // Not yet appeared in proven sector; check for timeout.
+                    if state.sector_start_epoch.is_none() {
+                        assert!(
+                            rt.curr_epoch() >= deal.start_epoch,
+                            "if sector start is not set, we must be in a timed out state"
+                        );
+
+                        let slashed = st
+                            .process_deal_init_timed_out(
+                                rt.store(),
+                                &mut et,
+                                &mut lt,
+                                id,
+                                &deal,
+                                state,
+                            )
+                            .unwrap();
+                        amount_slashed += slashed;
+
+                        if deal.verified_deal {
+                            timed_out_verified_deals.push(deal.clone());
+                        }
+                    }
+
+                    let (slash_amount, next_epoch) = st
+                        .update_pending_deal_state(
+                            rt.store(),
+                            state,
+                            deal,
+                            id,
+                            &mut et,
+                            &mut lt,
+                            rt.curr_epoch(),
+                        )
+                        .unwrap();
+                    amount_slashed += slash_amount;
+
+                    if next_epoch.is_some() {
+                        assert!(next_epoch.unwrap() > rt.curr_epoch());
+
+                        // TODO: can we avoid having this field?
+                        state.last_updated_epoch = OptionalEpoch(Some(rt.curr_epoch()));
+
+                        states
+                            .set(id, state)
+                            .map_err(|e| {
+                                ActorError::new(
+                                    ExitCode::ErrPlaceholder,
+                                    format!("failed to get deal: {}", e),
+                                )
+                            })
+                            .unwrap();
+                        let idx = next_epoch.unwrap();
```

Comment:
can this just be moved above to not do the unwrap twice? (also optionally you could do an `if let` here no not have to do this